src/System.Text.Json/tests/Serialization/Object.WriteTests.cs

```diff
@@ -53,5 +53,27 @@ public class ObjectObject
         {
             public object Object { get; set; }
         }
+
+        [Fact]
+        public static void WriteObject_PublicIndexer()
+        {
+            var indexer = new Indexer();
+            indexer[42] = default;
```

Comment:
Consider setting it to something other than 0.

Reply:
right.